### PR TITLE
Cleanup of the frontend code

### PR DESCRIPTION
**Pull request by paskal**
- replace undocumented `substr` with `substring`
- remove unused code
- inline a few variables
- simplify ifs when possible
- improve saveCollapsedComments documentation
- cleanup the unused imports
- remove unused variables and types

### CODE REVIEW

File: frontend/apps/remark42/app/components/comment-form/comment-form.tsx [hunk 1]
```diff
@@ -159,8 +159,7 @@ export class CommentForm extends Component<Props, State> {
       this.setState({
         isDisabled: false,
         isErrorShown: true,
-        // @ts-ignore
-        errorMessage: extractErrorMessageFromResponse(e, this.props.intl),
+        errorMessage: extractErrorMessageFromResponse(e as FetcherError, this.props.intl),
```
Comment (akellbl4) [inline frontend/apps/remark42/app/components/comment-form/comment-form.tsx:162]: to be honest assertion isn't better but we can let it be

File: frontend/apps/remark42/app/components/comment/comment-votes.tsx [hunk 2]
```diff
@@ -38,8 +38,7 @@ export function CommentVotes({ id, votes, vote, disabled, controversy = 0 }: Pro
       setErrorMessage(undefined);
       setTimeout(() => setLoadingState(null), 200);
     } catch (err) {
-      // @ts-ignore
-      setErrorMessage(extractErrorMessageFromResponse(err, intl));
+      setErrorMessage(extractErrorMessageFromResponse(err as FetcherError, intl));
```
Comment (akellbl4) [inline frontend/apps/remark42/app/components/comment/comment-votes.tsx:41]: Same here, we should check type of error before passing in into extract function like `err instanceof FectherError`

File: frontend/apps/remark42/app/components/comment/comment.tsx [hunk 3]
```diff
@@ -175,16 +174,6 @@ export class Comment extends Component<CommentProps, State> {
...
-  onBlockUserClick = (evt: Event) => {
-    const target = evt.currentTarget;
-
-    if (target instanceof HTMLOptionElement) {
-      // we have to debounce the blockUser function calls otherwise it will be
-      // called 2 times (by change event and by blur event)
-      this.blockUser(target.value as BlockTTL);
-    }
-  };
-
```
Comment (akellbl4) [inline frontend/apps/remark42/app/components/comment/comment.tsx:187]: 👍

File: frontend/apps/remark42/app/components/settings/settings.tsx [hunk 4]
```diff
@@ -91,8 +91,7 @@ class SettingsComponent extends Component<Props, State> {
   };
 
   __isUserHidden = (user: User): boolean => {
-    if (this.state.unhiddenUsers.indexOf(user.id) === -1) return true;
-    return false;
+    return this.state.unhiddenUsers.indexOf(user.id) === -1;
```
Comment (akellbl4) [inline frontend/apps/remark42/app/components/settings/settings.tsx:94]: let's use `includes` which returns `boolean`

File: frontend/apps/remark42/app/typings/global.d.ts [hunk 5]
```diff
@@ -34,18 +34,5 @@ declare global {
     };
   }
 
-  namespace NodeJS {
-    interface Global {
-      Headers: typeof Headers;
-      localStorage: typeof Storage;
-    }
-  }
+  namespace NodeJS {}
```
Comment (akellbl4) [inline frontend/apps/remark42/app/typings/global.d.ts:37]: we can remove it completely

File: frontend/apps/remark42/app/utils/loadLocale.ts [hunk 6]
```diff
@@ -55,7 +55,9 @@ export async function loadLocale(locale: string): Promise<Record<string, string>
     return import(/* webpackChunkName: "ar" */ '../locales/ar.json').then((res) => res.default).catch(() => enMessages);
   }
   if (locale === 'zh-tw') {
-    return import(/* webpackChunkName: "zh-tw" */ '../locales/zh-tw.json').then((res) => res.default).catch(() => enMessages);
+    return import(/* webpackChunkName: "zh-tw" */ '../locales/zh-tw.json')
+      .then((res) => res.default)
+      .catch(() => enMessages);
```
Comment (akellbl4) [inline frontend/apps/remark42/app/utils/loadLocale.ts:60]: this file is autogenerated we shouldn't do any formatting here

File: frontend/apps/remark42/package.json [hunk 7]
```diff
@@ -110,6 +110,7 @@
     "postcss-loader": "^7.0.0",
     "postcss-preset-env": "^7.7.2",
     "prettier": "^2.7.1",
+    "process": "^0.11.10",
```
Comment (akellbl4) [inline frontend/apps/remark42/package.json:113]: `process` is global variable, we don't need this package

**Review by akellbl4 — CHANGES_REQUESTED** [review]
@paskal I like those changes. I wrote few points which we should fix before merging it.

**Comment by akellbl4** [comment]
I just realized that you tried to address https://github.com/umputun/remark42/issues/1321. I'd say it's less than ideal way of addressing it, so I'd rather leave `ts-ignore` because assertions is nothing better that ignores because types are still unsafe.

**Review by akellbl4 — APPROVED** [review]
@paskal nice job!

**Review by umputun — APPROVED** [review]

**Pull request merged**